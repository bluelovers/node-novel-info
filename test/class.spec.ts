import fs from 'fs-extra';
import { join } from 'path';
import NodeNovelInfo from '../class';

describe(`README2.md`, () =>
{
	let md = fs.readFileSync(join(__dirname, 'res', 'README2.md'));

	let actual = NodeNovelInfo.createFromString(md);

	test(`createFromString`, () =>
	{
		expect(actual).toMatchSnapshot();

	});

	test(`title`, () =>
	{
		expect(actual.title()).toMatchSnapshot();

	});

	test(`titles`, () =>
	{
		expect(actual.titles()).toMatchSnapshot();

	});

	test(`illusts`, () =>
	{
		expect(actual.illusts()).toMatchSnapshot();

	});

	test(`authors`, () =>
	{
		expect(actual.authors()).toMatchSnapshot();

	});

	test(`sources`, () =>
	{
		expect(actual.sources()).toMatchSnapshot();

	});

	test(`sites`, () =>
	{
		expect(actual.sites()).toMatchSnapshot();
	});

})
