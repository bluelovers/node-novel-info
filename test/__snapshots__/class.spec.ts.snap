// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`README2.md authors 1`] = `
Array [
  "流優",
]
`;

exports[`README2.md createFromString 1`] = `
Object {
  "contribute": Array [
    "-妖妖幻想",
    "gasai谢丽尔",
    "二階堂真紅",
    "土之女神",
    "梨衣熊大好",
    "猫也学外语",
    "萌與幻想",
  ],
  "link": Array [
    "[dip.jp](https://narou.nar.jp/search.php?text=n2671do&novel=all&genre=all&new_genre=all&length=0&down=0&up=100) - 小説家になろう　更新情報検索",
    "[想自由生活的魔王默示录吧](https://tieba.baidu.com/f?kw=%E6%83%B3%E8%87%AA%E7%94%B1%E7%94%9F%E6%B4%BB%E7%9A%84%E9%AD%94%E7%8E%8B%E9%BB%98%E7%A4%BA%E5%BD%95&ie=utf-8 \\"想自由生活的魔王默示录\\")",
    "[【魔王默示录 - 整合】](https://tieba.baidu.com/p/5822195807 \\"【魔王默示录 - 整合】\\")",
    "[comic-walker](https://comic-walker.com/contents/detail/KDCW_FS01200334010000_68/)",
  ],
  "novel": Object {
    "author": "流優",
    "cover": "https://images-na.ssl-images-amazon.com/images/I/91r6R0%2B9L5L.jpg",
    "date": "2018-10-29T08:11:00+08:00",
    "preface": "注意到的時候，就已經成為了迷宮防衛機構的【魔王】了。周圍有強大的魔物徘徊著，因為貌似是為了生存競爭而拼命戰鬥的不太平的世界，所以為了不會死而發展迷宮來保護自身的，本應是如此的，然而為何會比想象的更加地安心療愈。……怎麼會變成這樣。

為了在那樣的世界活下去的主人公優紀（ユキ），發揮著那粗神經，與住在迷宮內自墮落的霸龍少女以及吸血鬼幼女一起，時而歡笑，時而生氣，時而尋求著浪漫，時而讓人會心一笑，隨心所欲地過著日子……

　なんか気が付いたら、ダンジョンの防衛機構である『魔王』になっていた。周囲には強大な魔物が徘徊し、生存競争を戦い抜く割と物騒な世界らしいので、死なないようにダンジョンを発展させて身を守るはずが、何故か思っていた以上にほのぼのしている。……どうしてこうなった。
　そんな世界を生きることになった主人公ユキは、その図太い神経を発揮して、ダンジョンに住み着いた自堕落な覇龍の少女や吸血鬼幼女らと共に、時に笑い、時に怒り、時にロマンを求め、時に頰をにやけさせながら、日々を好きに生きて行く……。

　作者ツイッター：『Ryuyu_』。 更新は週に二回か三回程。大体ほのぼの&コメディで進めていきます。
　書籍化しました。カドカワＢＯＯＫＳ様より、１～３巻が発売しています。やったぜ。
　コミカライズしました！　ドラドラドラゴンエイジ様と、コミックウォーカー様にて連載しております！",
    "publisher": "syosetu",
    "series": Object {
      "name": "魔王になったので、ダンジョン造って人外娘とほのぼのする",
    },
    "source": "http://ncode.syosetu.com/n2671do/",
    "status": "連載",
    "tags": Array [
      "R15",
      "node-novel",
      "syosetu",
      "ほのぼの",
      "ステータス",
      "ダンジョン",
      "ダンジョンマスター",
      "ダンジョン運営",
      "人外",
      "人外娘",
      "人外転生",
      "幼女",
      "幼女ホイホイ",
      "恋愛",
      "擬人化",
      "書籍化",
      "残酷な描写あり",
      "男主人公",
      "異世界転生",
      "魔王",
    ],
    "title": "魔王になったので、ダンジョン造って人外娘とほのぼのする",
    "title_zh": "想自由生活的魔王默示錄",
  },
  "options": Object {
    "downloadOptions": Object {
      "filePrefixMode": 4,
      "noFilePadend": true,
      "startIndex": 1,
    },
    "syosetu": Object {
      "novel_id": "n2671do",
      "txtdownload_id": "922579",
    },
    "textlayout": Object {
      "allow_lf2": true,
    },
  },
}
`;

exports[`README2.md illusts 1`] = `Array []`;

exports[`README2.md sites 1`] = `
Array [
  Object {
    "data": Object {
      "novel_id": "n2671do",
      "txtdownload_id": "922579",
    },
    "site": "syosetu",
  },
]
`;

exports[`README2.md sources 1`] = `
Array [
  "http://ncode.syosetu.com/n2671do/",
]
`;

exports[`README2.md title 1`] = `"想自由生活的魔王默示錄"`;

exports[`README2.md titles 1`] = `
Array [
  "魔王になったので、ダンジョン造って人外娘とほのぼのする",
  "想自由生活的魔王默示錄",
]
`;
